### ISSUE TITLE
Change $cache_translations to protected or add method setCachedTranslations(array)

### ISSUE THREAD

**Issue by marcing** [phpmyadmin/motranslator#30]
Hi!

I am trying to implement in-memory cache for motranslator, but I noticed I have to loop the whole array and call setTranslation() for each msgid. Maybe you could simplify it by changing cache_translations to protected or by adding another method for setting all translations at once?

```php
class CachedTranslator extends Translator
{
	/**
	 * @param string $filename
	 */
	public function __construct($filename)
	{
		// TODO: read from cache, set $cached_translations and return or let it continue with reading the file on cache miss.
		parent::__construct($filename);
	}
}
```
Thank you!

**Comment by marcing** [comment]
Actually there is one more issue. There is no way to access cached translations for extending class.
There should be also a method getCachedTranslations() to implement cache.

**Comment by williamdes** [comment]
@mauriciofauth what do you think about this issue ?

**Comment by williamdes** [comment]
@mauriciofauth did you miss this notification?

**Comment by MauricioFauth** [comment]
> @mauriciofauth did you miss this notification?

Yes.

> what do you think about this issue ?

I think we could add the `setTranslations` and `getTranslations` methods.

**Comment by williamdes** [comment]
https://github.com/phpmyadmin/motranslator#translator-api-usage

**Comment by williamdes** [comment]
Released as 5.2.0 :rocket:

**Comment by marcing** [comment]
Thanks!
